### ISSUE TITLE
[BUG]  ModuleNotFoundError: No module named 'sktime.utils.data_processing'

### ISSUE THREAD

**Issue by albas99** [sktime/sktime#4509]
I'm trying to import the function `from_2d_array_to_nested` using `from sktime.utils.data_processing import from_2d_array_to_nested` but it kept giving me the error `
ModuleNotFoundError: No module named 'sktime.utils.data_processing'`. I checked the documentation and the entire object `sktime.utils.data_processing` doesn't exist even though a lot of solutions online seem to be saying that's the correct import. There's also nothing in the documentation that talked about whether it's been moved or removed entirely.

sktime version: 0.17.2

**Comment by fkiraly** [comment]
@albas99, we haven't had this module in a while - where do you have the line from?

Since a year or more, the data format conversion utilities are in `sktime.datatypes`, you probably want to use `convert_to("nested_univ")`, although we recommend to use the `pd-multiindex` format rather than `nested_univ`.

For more details on data formats and conversions, check out the tutorial numbered "AA" (for appendix A) on binder:
https://mybinder.org/v2/gh/sktime/sktime/main?filepath=examples

**Comment by albas99** [comment]
The code is from [here](https://github.com/MVRonkin/Time-Series-Analysis-Lectures-and-Workshops), seems like most of it is outdated.

**Comment by fkiraly** [comment]
I see - it looks like the repository is still being maintained, maybe open an issue there?

Happy to help updating the code if I know precisely where it's located.

Further, I ran a search on the `sktime` documentation just to check whether we are still pointing to the module somewhere, but it does not look like to be the case.

**Comment by fkiraly** [comment]
(based on this I'd classify this as not a bug, but let's keep it open until the discussion is concluded)

**Comment by albas99** [comment]
> I see - it looks like the repository is still being maintained, maybe open an issue there?

Yeah, the repository is still being maintained but only the files, not the code. I'm planning on going through the code and updating the imports and deprecated functions but this one particularly has been driving me crazy. I've tried your suggestion of `convert_to("nested_univ")` and `pd-multiindex` but none of them seem to have fixed the issue. 

> Happy to help updating the code if I know precisely where it's located.

Yeah I'd greatly appreciate that. Here's the actual notebook: https://github.com/MVRonkin/Time-Series-Analysis-Lectures-and-Workshops/blob/main/workshops/WS_6.ipynb, in the 5th cell.

**Comment by fkiraly** [comment]
Ok, @albas99, I've made a PR here:
https://github.com/MVRonkin/Time-Series-Analysis-Lectures-and-Workshops/pull/1

This should update the failing cells to the newest `sktime`.

Note that there is no CI or similar on the repository, so you would still have to test whether it works.

**Comment by albas99** [comment]
@fkiraly, Wow, that was very fast🤯. Thank you very much for that. I'll talk to him so he can merge it as soon as possible.

**Comment by fkiraly** [comment]
if you would be so kind to test whether the notebook runs for you, that would be great

**Comment by fkiraly** [comment]
If you're interested in tutorials for classification, clustering etc in general, you may like to attend or earmark the "official" tutorial at pydata London 2023: https://london2023.pydata.org/cfp/talk/LAAP8N/

even if you cannot attend, we'll record and publish the recording usually a few months after.

**Comment by albas99** [comment]
> if you would be so kind to test whether the notebook runs for you, that would be great

I'm not the owner of the repo, it belongs to my professor so I'm also waiting for him to merge the changes so I can pull them and test it but judging from the pull request everything seem to work fine.



> If you're interested in tutorials for classification, clustering etc in general, you may like to attend or earmark the "official" tutorial at pydata London 2023: https://london2023.pydata.org/cfp/talk/LAAP8N/
> 
> even if you cannot attend, we'll record and publish the recording usually a few months after.
> 
> There's also this one obviously: https://github.com/sktime/sktime/blob/main/examples/02_classification.ipynb

Unfortunately I can't attend that in-person, though I'd love to if there's a virtual version as well. I just came across the sktime internship though which I think is something I'm definitely applying for.
